image.go

```diff
@@ -0,0 +1,55 @@
+package canvas
+
+import (
+	"bytes"
+	"image"
+	"image/jpeg"
+	"image/png"
+	"io"
+)
+
+type ImageMimetype int
```

Comment:
I agree this would be a good improvement, but I would press to put that in a new PR where you change this package-wide. Not just for images, but for all mimetypes. In the mean time let's keep this a simple string. Please note, even though I wasn't aware of this and it is/was wrong in the code, that `image/jpg` doesn't exist and it should be `image/jpeg`. There should thus be no confusion.

Reply:
In both cases it would be a breaking change :
- image dedicated type now; general dedicated type later (current state)
- `string` now; dedicated type later (your proposition)

I would personnally rather have the first breaking change, since removing the `ImageMimetype` type will be detected at compile time (where the `string` could split through in some cases).

A replace-all is also easier (the string `"image/jpeg"` could be a mimetype or an import path...)

Reply:
But using `ImageMimetype` would require a breaking change as well in the future (into using a package-wide `Mimetype`)! To be honest I don't see the big advantage is of using an `int` here. Yes it compares faster than a string, but it also requires more code! And what other benefits does it bring? People can still use arbitrary integers, the list is by no means excluding other values much like a string. Additionally, thinking the bigger picture, the SVG renderer should also be using the raw bytes if it can, but as you only added JPEG and PNG, this prevents users from using TIFF, GIF, BMP, or WebP files or other less-known file formats. And what about the font mimetypes for `font.go`?

When using the `NewJPEGImage` and `NewPNGImage` functions, converting from string to enum is actually not a breaking change. So I don't see a downside about thinking this through and implementing this right package-wide. It would require `mimetype.go` file that can parse mimetypes (this is not trivial) and convert to their canonical string form. It is a good idea in principle what you are advocating, but I stress that we put this in a separate PR please.

Reply:
All right, I get your points.

I removed the `ImageMimetype` and used a string everywhere :smile: